### ISSUE TITLE
Login with Username/Password

### ISSUE THREAD

**Issue by utzelberlin** [cioraneanu/firefly-pico#129]
I would really like to use Pico. But unfortunately it doesn't help me with the daily manual copying and pasting of the Personal Token. I would be very grateful for a solution.

Reference: https://github.com/cioraneanu/firefly-pico/issues/31#issuecomment-2104305263_

I'm sorry but i cannot be satisfied with this solution. I handle the data of my browser as deleted by close him. That's for security reasons.

The Personal Token is not really manageable. To recognize a user, there is the username/password concept and every browser can handle it (e.g. through AutoFill if I wish). This is not possible with a Personal Token I have to set every time I want to use Pico and the benefits of Pico (Ex. quick recording of transactions) are lost before I could start.

I also can't see why adjustments to the Firefly database should be necessary. Only one Pico table is required that connects username/password with the Pico Account Settings including a Personal Token. Or the 2 fields username/password in the Account Settings Table to find the right assignment.

With this solution your wife, you self and also I and all other Users could use Pico without limitations I think. A checkbox "remember login" on the top could even avoid logging in for users who don't want to use it. Isn't it like that?

**Comment by cioraneanu** [comment]
Hi.
Thank you for your feedback. As a recomandation don't copy paste the same comment in multiple places 😇 

Unfortunetly this is not something that is easy to implement nor is it useful to a large number of users and there are ways around it like using a different browser which doesn't clear localStorage for FireflyPico.

Some of the problems with your suggestion:
- Saving username + password in Pico's database would create 2 sources of truth. Imagine changing the password inside Firefly, Pico would not know about it.
- You'll have to create new UI pages in Pico to manage these credentials
- There is no Firefly API for username + password. You would probably have to mimick the login mechanism from Firefly which probaly uses cookies.

In short the drawbacks far exceed the benefits.

**Comment by utzelberlin** [comment]
Many thanks for your reply!

Maybe you're misunderstanding me. There is no reason to exchange username and password with Firefly. The connection to Firefly remains unchanged with personal tokens only. 

In Pico we just need a login window. After entering your username and password, the appropriate personal token will be determined, which will be stored in the DB in the future. From there everything runs unchanged as before.

Best regards

**Comment by cioraneanu** [comment]
I'm sorry but this still adds another layer of complexity without any real benefits. This makes it even more confusing for new users since it's not common to have a username and password if you already have a JWT.

The reality is you've created a problem that has easy solutions but you're looking at complex work-arounds. 😄

**Comment by utzelberlin** [comment]
Your answers just shows that you didn't understand the problem at all. but it's your app and if there's no interest, I have to accept it that way.

**Comment by gigabites19** [comment]
> Your answers just shows that you didn't understand the problem at all. but it's your app and if there's no interest, I have to accept it that way.

Tone of this comment is tone-deaf. It's an open source project, author has no obligations to you or anyone else. You're free to modify it however you want, that's the beauty of it.
